lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1325,6 +1326,19 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
                            nominalType)
             .fixItRemove(DS->getSourceRange());
         diagnosed = true;
+      // if the type is public and not frozen, then the method must not be
```

Comment:
This should be replaced with an isResilient() query that passes in the current DeclContext’s getResilienceExpansion()

Reply:
I specifically *don't* want to hang it on resilience, since we shouldn't make internal type layout a part of the API contract regardless of whether someone is using library evolution or not.

Reply:
In that case, you might be able to use `isFormallyResilient()`, which we use for enum exhaustiveness IIRC.